examples/2d/move_sprite.rs

```diff
@@ -1,46 +1,54 @@
-//! Renders a 2D scene containing a single, moving sprite.
+//! Renders a 2D scene in which the *Bevy* logo is rendered as a sprite, moving up and down.
 
 use bevy::prelude::*;
 
 fn main() {
     App::new()
         .add_plugins(DefaultPlugins)
         .add_startup_system(setup)
-        .add_system(sprite_movement)
+        .add_system(update_moveables)
+        .add_system(update_birds)
         .run();
 }
 
+/// A component for any sprite that might move.
 #[derive(Component)]
-enum Direction {
-    Up,
-    Down,
-}
+struct Velocity(Vec2);
```

Comment:
We are deriving `Deref` and `DerefMut` to clean up the `.0` spam in (some) other examples. Should we do the same here?

Reply:
I also thought about that. It would add ~~10 lines of code (guess!) to a 50-line example, though. Would that really be worth it?

Reply:
It wouldn't add any lines of code
```diff
diff --git a/examples/2d/move_sprite.rs b/examples/2d/move_sprite.rs
index 5290ac1e2..88b78f1bb 100644
--- a/examples/2d/move_sprite.rs
+++ b/examples/2d/move_sprite.rs
@@ -12,7 +12,7 @@ fn main() {
 }
 
 /// A component for any sprite that might move.
-#[derive(Component)]
+#[derive(Component, Deref, DerefMut)]
 struct Velocity(Vec2);
 
 /// A marker component for our particular sprite.
@@ -36,7 +36,7 @@ fn setup(mut commands: Commands, asset_server: Res<AssetServer>) {
 fn update_moveables(time: Res<Time>, mut query: Query<(&Velocity, &mut Transform)>) {
     let delta = time.delta_seconds();
     for (velocity, mut transform) in &mut query {
-        transform.translation += velocity.0.extend(0.0) * delta;
+        transform.translation += velocity.extend(0.0) * delta;
     }
 }
 
@@ -49,7 +49,7 @@ fn update_moveables(time: Res<Time>, mut query: Query<(&Velocity, &mut Transform
 fn update_birds(mut query: Query<(&mut Velocity, &Transform), With<Bird>>) {
     for (mut v, trf) in &mut query {
         if trf.translation.y.abs() > 200. {
-            v.0 *= -1.;
+            **v *= -1.;
         }
     }
 }
 ```

Reply:
Oh gosh. I never `#[derive…]` those in my own code-base so I totally forgot you *could*.

It's definitely worth doing, then.

Reply:
One other thing: You call `.extend(0.0)` – which matches my own style for `f32`s (`.0`) – but I had originally copied the original examples style (`0.`). Which one is preferred?
(If I had a say, I'd go with `.0` but then I would also patch all the other `f32` literals in the file.)

Reply:
That's not my code, but I prefer `0.`.

Reply:
At a glance it seems like we're generally doing `0.0`

Reply:
Ok. I added `Deref` and `DerefMut` as the latest commit to the branch but... the result isn't very satisfying.

I don't like the double-dereference-mul-asign idiom in your patch – that would be a show-stopper for any non-Rustacean reader in otherwise trivially easy code – so I left that out... but... yeah. I'm not sold on the end-product.